### PR TITLE
Avgpool revise

### PR DESCRIPTION
**Pull request by pszmel**
1. Update the spec
2. Create unit-tests
3. Remove second constructor and add default values to the first one
4. Add type_prop tests for default values
5. Add 5D input single layer test instances
6. Update validate_and_infer_types method
7. Update infer_batched_pooling_forward and try_apply_auto_padding methods
8. Update auto_padding_spatial_dims_dynamic type_prop test for binary_conv, conv, deformable_conv, group_conv and max_pool

Changes in infer_batched_pooling_forward and try_apply_auto_padding required me to change the expected pads_begin and pads_end in 1 type_prop test for ops listed in point 8.

ticket: 37422

### CODE REVIEW

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/pooling/AvgPool_1.md [hunk 1]
```diff
@@ -68,26 +69,87 @@
...
 
-**Example**
+**Examples**
 
 ```xml
 <layer ... type="AvgPool" ... >
-        <data auto_pad="same_upper" exclude-pad="true" kernel="3,3" pads_begin="0,0" pads_end="1,1" strides="2,2"/>
-        <input> ... </input>
-        <output> ... </output>
+        <data auto_pad="same_upper" exclude-pad="true" kernel="2,2" pads_begin="0,0" pads_end="1,1" strides="2,2"/>
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:99]: `exclude-pad` -> `exclude_pad`.

File: docs/ops/pooling/AvgPool_1.md [hunk 2]
```diff
@@ -68,26 +69,87 @@
...
+        <port id="1">
+            <dim>1</dim>
+            <dim>3</dim>
+            <dim>32</dim>
+        </port>
+    </output>
+</layer>
+
+<layer ... type="AvgPool" ... >
+        <data auto_pad="explicit" exclude-pad="true" kernel="2,2" pads_begin="1,1" pads_end="1,1" strides="2,2"/>
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:117]: wrong indentation

File: docs/ops/pooling/AvgPool_1.md [hunk 3]
```diff
@@ -68,26 +69,87 @@
...
+            <dim>1</dim>
+            <dim>3</dim>
+            <dim>17</dim>
+            <dim>17</dim>
+        </port>
+    </output>
+</layer>
+
+<layer ... type="AvgPool" ... >
+        <data auto_pad="valid" exclude-pad="true" kernel="2,2" pads_begin="1,1" pads_end="1,1" strides="2,2"/>
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:137]: Wrong indentation.

File: docs/ops/pooling/AvgPool_1.md [hunk 4]
```diff
@@ -68,26 +69,87 @@
...
 
 \f[
 output_{j} = \frac{\sum_{i = 0}^{n}x_{i}}{n}
 \f]
 
-**Example**
+**Examples**
 
 ```xml
 <layer ... type="AvgPool" ... >
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:98]: All cases are for `exclude_pad="true"`. Please, provide examples where `exclude_pad="false"`

File: docs/ops/pooling/AvgPool_1.md [hunk 5]
```diff
@@ -68,26 +69,87 @@
...
+            <dim>1</dim>
+            <dim>3</dim>
+            <dim>32</dim>
+          </port>
+        </input>
+        <output>
+        <port id="1">
+            <dim>1</dim>
+            <dim>3</dim>
+            <dim>32</dim>
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:111]: wrong dimension

File: docs/ops/pooling/AvgPool_1.md [hunk 6]
```diff
@@ -68,26 +69,87 @@
...
   * **Required**: *no*
   * **Note**: *pads_begin* and *pads_end* attributes are ignored when *auto_pad* is specified.
 
 **Inputs**:
 
-*   **1**: 4D or 5D input tensor. Required.
+*   **1**: 3D, 4D or 5D input tensor. Required.
+
+**Outputs**:
+  * **1**: Input shape can be either (N,C,H), (N,C,H,W) or (N,C,H,W,D). Then the corressponding output shape will be (N,C,H_out), (N,C,H_out,W_out) or (N,C,H_out,W_out,D_out)
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:81]: Change "(N,C,H)" to "`[N, C, H]`" (use back tick characters "`").  The same for other dimensions.

File: docs/ops/pooling/AvgPool_1.md [hunk 7]
```diff
@@ -68,26 +69,87 @@
...
+       
+        H_out = (H + pads_begin[0] + pads_end[0] - kernel[0] / strides[0]) + 1
+        
+        W_out = (H + pads_begin[1] + pads_end[1] - kernel[1] / strides[1]) + 1
+        
+        D_out = (H + pads_begin[2] + pads_end[2] - kernel[2] / strides[2]) + 1
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:87]: Move this code to the "Detailed description section."

File: docs/ops/pooling/AvgPool_1.md [hunk 8]
```diff
@@ -68,26 +69,87 @@
...
 
 **Inputs**:
 
-*   **1**: 4D or 5D input tensor. Required.
+*   **1**: 3D, 4D or 5D input tensor. Required.
+
+**Outputs**:
+  * **1**: Input shape can be either (N,C,H), (N,C,H,W) or (N,C,H,W,D). Then the corressponding output shape will be (N,C,H_out), (N,C,H_out,W_out) or (N,C,H_out,W_out,D_out)
+       
+        H_out = (H + pads_begin[0] + pads_end[0] - kernel[0] / strides[0]) + 1
```
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:83]: Surround formula with back tick characters "`"

File: docs/ops/pooling/AvgPool_1.md (hunk 6 above)
Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:81]: "corressponding" -> "corresponding"

Comment (lazarevevgeny) [inline docs/ops/pooling/AvgPool_1.md:81]: "shape will be" -> "shape is"

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by mbencer — APPROVED** [review]

**Comment by mbencer** [comment]
Pooling from ONNX supports 1D and 2D cases. How to deal with it? Do we need new ops in new version?
@lazarevevgeny What do you think?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: ngraph/test/type_prop/convolution.cpp [hunk 9]
```diff
@@ -2638,10 +2638,9 @@ TEST(type_prop, conv_v1_partial_auto_padding_same_spatial_dims_dynamic)
     auto conv = make_shared<op::v1::Convolution>(
         data_batch, filters, strides, pads_begin, pads_end, dilations, auto_pad);
 
-    ASSERT_TRUE(conv->get_output_partial_shape(0).same_scheme(
-        {1, 1, Dimension::dynamic(), Dimension::dynamic()}));
-    ASSERT_EQ(conv->get_pads_begin(), (CoordinateDiff{}));
-    ASSERT_EQ(conv->get_pads_end(), (CoordinateDiff{}));
+    ASSERT_TRUE(conv->get_output_partial_shape(0).same_scheme({1, 1, Dimension::dynamic(), 5}));
+    ASSERT_EQ(conv->get_pads_begin(), (CoordinateDiff{0, 1}));
+    ASSERT_EQ(conv->get_pads_end(), (CoordinateDiff{0, 1}));
```
[the remark this answers is not in the document]
Comment (pszmel) [reply ngraph/test/type_prop/convolution.cpp:2643]: I wrote about that in the description section

**Review by lazarevevgeny — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**